src/actions.ml

```diff
@@ -628,76 +672,536 @@ let minimize_failed_tests ~bot_info ~owner ~repo ~pr_number ~base ~head
+              Lwt.return_error
+                (f "Couldn't find pipeline check summary for base commit %s."
+                   base)
+          | ([], _), (_, _) ->
+              Lwt.return_error
+                (f "Couldn't find pipeline check for base commit %s." base)
+          | (_ :: _ :: _, _), (_, _) ->
+              Lwt.return_error
+                (f
+                   "Found several pipeline checks instead of one for base \
+                    commit %s."
+                   base) ) )
+
+type ci_minimization_request =
+  | Auto
+  | RequestSuggested
+  | RequestAll
+  | RequestExplicit of string list
+
+type ci_minimization_suggestion_kind =
+  (* this is a job that we suggest minimization for *)
+  | Suggested
+  (* we don't suggest running this job, but it's okay to; give a reason that we don't suggest it *)
+  | Possible of string
+  (* we expect that running this job won't work; give a reason *)
+  | Bad of string
+
+(* For grammatical correctness, all messages are expected to follow "because" *)
+let ci_minimization_suggest ~base
+    { base_job_failed
+    ; base_job_errored
+    ; head_job_succeeded
+    ; missing_error
+    ; non_v_file
+    ; job_kind
+    ; overlayed } =
+  if head_job_succeeded then Bad "job succeeded!"
+  else if missing_error then Bad "no error message was found"
+  else
+    match (base_job_errored, non_v_file) with
+    | _, Some filename ->
+        Bad (f "error message did not occur in a .v file (%s)" filename)
+    | Some err, _ ->
+        Possible (f "base job at %s errored with message %s" base err)
+    | None, None ->
+        if base_job_failed then Possible (f "base job at %s failed" base)
+        else if overlayed then Possible (f "an overlay is present")
+        else if
+          not (List.exists ~f:(String.equal job_kind) ["library"; "plugin"])
+        then
+          Possible
+            (f "the job is a %s which is not a library nor a plugin" job_kind)
+        else Suggested
+
+type ci_pr_minimization_suggestion =
+  | Suggest
+  | RunAutomatically
+  | Silent of string
+
+let suggest_ci_minimization_for_pr = function
+  (* don't suggest if there are failed test-suite jobs (TODO: decide about async?) *)
+  | {failed_test_suite_jobs= _ :: _ as failed_test_suite_jobs} ->
+      Silent
+        (f "the following test-suite jobs failed: %s"
+           (String.concat ~sep:", " failed_test_suite_jobs))
+  (* This next case is a dummy case so OCaml doesn't complain about us
+     never using RunAutomatically; we should probably remove it when
+     we add a criterion for running minimization automatically *)
+  | {labels}
+    when List.exists ~f:(String.equal "coqbot request ci minimization") labels
+    ->
+      RunAutomatically
+  (* TODO: Should we suggest on draft? *)
+  | {labels} when List.exists ~f:(String.equal "kind: infrastructure") labels ->
+      Silent "this PR is labeled with kind: infrastructure"
+  (* TODO: Should we suggest on draft? *)
```

Comment:
Hum, yeah, I would think so. If the test-suite passed on a draft PR but a job failed, then this means the test-suite should be expanded, doesn't it?